File: benchmarks/executable/restaurant_integer_widening.c
// Expected exit: 0
// Expected stdout:
// Widen -7/200/202/203
#include <stdint.h>
#include <stdio.h>

#ifdef _WIN32
#include <fcntl.h>
#include <io.h>
#endif

static volatile int8_t signed_input = -7;
static volatile uint8_t first_unsigned_input = 200u;
static volatile uint8_t binding_input = 202u;
static volatile uint8_t alias_input = 203u;

int main(void) {
#ifdef _WIN32
  if (_setmode(_fileno(stdout), _O_BINARY) == -1) return 1;
#endif
  const int16_t returned = signed_input;
  const int16_t called = first_unsigned_input;
  const uint16_t unsigned_value = binding_input;
  const int64_t alias_value = alias_input;
  return printf("Widen %d/%d/%u/%lld\n", (int)returned, (int)called,
                (unsigned)unsigned_value,
                (long long)alias_value) < 0;
}

File: benchmarks/executable/restaurant_integer_widening.rs
// Expected exit: 0
// Expected stdout:
// Widen -7/200/202/203

use std::hint::black_box;

fn main() {
    let returned: i16 = black_box(-7_i8).into();
    let called: i16 = black_box(200_u8).into();
    let unsigned: u16 = black_box(202_u8).into();
    let alias: i64 = black_box(203_u8).into();
    println!("Widen {returned}/{called}/{unsigned}/{alias}");
}
